server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -144,27 +144,37 @@ private static void update(ServerContext context, Mutation m, KeyExtent extent)
   public static void update(ServerContext context, ServiceLock zooLock, Mutation m,
       KeyExtent extent) {
     Writer t = extent.isMeta() ? getRootTable(context) : getMetadataTable(context);
-    update(context, t, zooLock, m);
+    update(context, t, zooLock, m, extent);
   }
 
   public static void update(ServerContext context, Writer t, ServiceLock zooLock, Mutation m) {
+    update(context, t, zooLock, m, null);
+  }
+
+  public static void update(ServerContext context, Writer t, ServiceLock zooLock, Mutation m,
+      KeyExtent extent) {
```

Comment:
I see there's only ever two places in the code that called the old method without the extent. The first is the one you changed above to call the new method, and where extent is never null. The second is in MetaConstraintRetryIT, which also has an extent that is non-null. I think there's no reason to have both update methods. You can just add extent to the existing one, without preserving the old behavior. You only have to update two references, and there's never a need to pass null into the method for extent, or to check for null in the log message where you have the ternary operator to change the method if it's null. If, for some reason it does end up being null, it's not a problem for it to say "for extent null".

Reply:
Cool. Fixed.